gdraw/ggdkdraw.c

```diff
@@ -592,7 +592,7 @@ static GWindow _GGDKDraw_CreateWindow(GGDKDisplay *gdisp, GGDKWindow gw, GRect *
             nw->is_dlg = true;
         } else if (!nw->is_dlg) {
             ++gdisp->top_window_count;
-        } else if (nw->restrict_input_to_me && gdisp->last_nontransient_window != NULL) {
+        } else if (nw->restrict_input_to_me && (gdisp->last_nontransient_window != NULL || gdisp->restrict_count > 0)) {
```

Comment:
I don't doubt the report that this fixes the associated issue. However, from the standpoint of "inspection" I'm concerned about the generality of the fix. We now apparently know there are windows that set `restrict_input_to_me` and aren't transient. What if one such window is created at a time when `last_nontransient_window` happens to be NULL (for whatever other strange reason)? In that case we'll have "grabbing" window 1 that doesn't wind up on the transient_stack. Then if another `restrict_input_to_me` window opens under these circumstances this guard could evaluate to false (because `last_nontransient_window` could still be null, and we wouldn't have incremented `restrict_count`. 

Doesn't it seem safter to have this guard just be `} else if (nw->restrict_input_to_me) { ` and then have `GGDKDrawSetTransientFor` get it into the `transient_stack` and adjust `restrict_count` in whatever way it needs to to avoid problems? If some of those windows aren't strictly transients, we could always rename the stack ...

Reply:
Alternative: Since `GGDKDrawSetTransientFor()` is a gdraw function pointer method, we could factor out the stack manipulation code into a separate static function and call that here and from `GGDKDrawSetTransientFor`. That would allow the naming to better match whatever we call this transient-or-restricted status. (I don't care all that much about names matching up in these complex legacy situations, but I won't discount the concern.)